openedx/features/offline_content/docs/001-mobile-offline-content-support.rst

```diff
@@ -0,0 +1,135 @@
+    * **Update archive**: Check generated archive creation date and update it if less than course publishing date.
+* **Implement a Mobile Local Storage Mechanism**: Use the device's local storage to save course content for offline access.
+    * **Extend blocks API**: Add links to download blocks content and  where it is possible.
+* **Sync Mechanism**: Periodically synchronize local data with the server when the device is online.
+    * **Sync on app side**: On course outline screen, check if the course content is up to date and update it if necessary.
+    * **Sync user responses**: When the device is offline, save user responses locally and send them to the server when the device is online.
+* **Selective Download**: Allow users to choose specific content to download for offline use.
+* **Full Course Download**: Provide an option to download entire courses for offline access.
+
+Supported xBlocks in offline mode
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It was decided to include a fraction of Open edX xBlocks to be supported.
+The following list of blocks is currently planned to be added to the support:
+
+* **Common problems**:
+    * **Checkboxes** - full support
+    * **Dropdown** - full support
+    * **Multiple Choice** - full support
+    * **Numerical Input** - full support
+    * **Text Input** - full support
+    * **Checkboxes with Hints and Feedback** - partial support without Hints and Feedback
+    * **Dropdown with Hints and Feedback** - partial support without Hints and Feedback
+    * **Multiple Choice with Hints and Feedback** - partial support without Hints and Feedback
+    * **Numerical Input with Hints and Feedback** - partially supported without Hints and Feedback
+    * **Text Input with Hints and Feedback** - partially supported without Hints and Feedback
+* **Text**:
+    * **Text** - full support
+    * **IFrame Tool** - full support
+    * **Raw HTML** - full support
+    * **Zooming Image Tool** - full support
+* **Video** - already supported
+
+
+Consequences
+------------
+
+* Enhanced learner experience with flexible access to course materials.
+* Increased accessibility for learners in regions with poor internet connectivity.
+* Improved engagement and completion rates due to uninterrupted access to content.
+* Potential increase in app size due to locally stored content.
+* Increased complexity in managing content synchronization and updates.
+* Need for continuous monitoring and updates to handle new content types and formats.
+
+Rejected Solutions
+------------------
+
+Store common .js and .css files of blocks in a separate folder:
+    * This solution was rejected because it is unclear how to track potential changes to these files and re-generate the content of the blocks.
```

Comment:
I am concerned that without any de-duping of JS, we will be saving and serving thousands of exactly-identical copies of the ProblemBlock's javascript and css.